pkg/logging/tmc.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package logging supplies common constants to ensure consistent use of structured logs.
+package logging
+
+const (
+	// SyncTargetKeyPrefix is the prefix used for all the keys related to a SyncTarget.
+	SyncTargetKeyPrefix = "syncTarget."
+
+	// SyncTargetWorkspaceKey is used to specify a workspace when a log is related to a SyncTarget.
+	SyncTargetWorkspaceKey = SyncTargetKeyPrefix + "workspace"
```

Comment:
In which cases does a client want to add these to the logger, but cannot call `logging.WithObject(logger, syncTarget)`?

Reply:
In all the places where a SyncTargetKey is required without having the Synctarget object at hand.
In addition, I only need to log the key, not the whole object (with apiVersion, etc ...)

Reply:
>  I only need to log the key, not the whole object (with apiVersion, etc ...)

That's what `WithObject()` does :)